proxysql/test-proxysql

```diff
@@ -38,45 +38,82 @@ function process_proxysql_output {
-
-if [[ "${PXC_VERSION}" == "PXC80" ]]; then
-    sudo wget -q -O ${WORKDIR_ABS}/WORKDIR/Percona-XtraDB-Cluster_8.0.20-11.3_Linux.x86_64.glibc2.12-minimal.tar.gz https://www.percona.com/downloads/Percona-XtraDB-Cluster-LATEST/Percona-XtraDB-Cluster-8.0.20-11.3/binary/tarball/Percona-XtraDB-Cluster_8.0.20-11.3_Linux.x86_64.glibc2.12-minimal.tar.gz
-    sudo tar -C ${WORKDIR_ABS}/WORKDIR/PXC --strip-components=1 -zxpf $(ls $WORKDIR_ABS/WORKDIR/Percona-XtraDB-Cluster_8.0.20-11.3_Linux.x86_64.glibc2.12-minimal.tar.gz | head -1)
-elif [[ "${PXC_VERSION}" == "PXC57" ]]; then
-    sudo wget -q -O ${WORKDIR_ABS}/WORKDIR/Percona-XtraDB-Cluster-5.7.31-rel34-45.3.Linux.x86_64.glibc2.12-minimal.tar.gz  https://www.percona.com/downloads/Percona-XtraDB-Cluster-57/Percona-XtraDB-Cluster-5.7.31-31.45-3/binary/tarball/Percona-XtraDB-Cluster-5.7.31-rel34-45.3.Linux.x86_64.glibc2.12-minimal.tar.gz
-    sudo tar -C ${WORKDIR_ABS}/WORKDIR/PXC --strip-components=1 -zxpf $(ls $WORKDIR_ABS/WORKDIR/Percona-XtraDB-Cluster-5.7.31-rel34-45.3.Linux.x86_64.glibc2.12-minimal.tar.gz | head -1)
-    if [ -f /usr/bin/yum ]; then
-        sudo yum remove -y Percona-Server-shared-56-5.6.49-rel89.0.1.el7.x86_64
-        if [[ $RHEL -eq 8 ]]; then
-            sudo yum remove -y mariadb-connector-c-config-3.0.7-1.el8.noarch
-        fi
-        sudo yum install -y percona-xtrabackup-24 Percona-XtraDB-Cluster-57 proxysql2
-    else
-        sudo apt update -y
-        sudo apt install -y percona-xtrabackup-24 percona-xtradb-cluster-client-5.7 proxysql2
-    fi
-fi
+rm -rf ${WORKDIR_ABS}/WORKDIR ${WORKDIR_ABS}/PROXY_ADMIN
+mkdir -p ${WORKDIR_ABS}/PROXY_ADMIN ${WORKDIR_ABS}/WORKDIR/proxysql-2.0/usr/bin ${WORKDIR_ABS}/WORKDIR/proxysql-2.0/etc
+
+if [ -f /usr/bin/yum ]; then
+# Installing proxysql 2.0.latest package
+  sudo percona-release enable proxysql testing
+  sudo yum install -y proxysql2
+
+  if [[ "${PXC_VERSION}" == "PXC80" ]]; then
+# Downloading PXC-8.0 tarball package
+    sudo wget -q -O ${WORKDIR_ABS}/WORKDIR/Percona-XtraDB-Cluster_8.0.22-13.1_Linux.x86_64.glibc2.17-minimal.tar.gz https://www.percona.com/downloads/Percona-XtraDB-Cluster-LATEST/Percona-XtraDB-Cluster-8.0.22/binary/tarball/Percona-XtraDB-Cluster_8.0.22-13.1_Linux.x86_64.glibc2.17-minimal.tar.gz
+
+# Installing PXC-8.0 client package
+    sudo percona-release enable pxc-80 release
+    sudo yum install -y percona-xtradb-cluster-client
+  elif [[ "${PXC_VERSION}" == "PXC57" ]]; then
+# Downloading PXC-5.7 tarball package
+    sudo wget -q -O ${WORKDIR_ABS}/WORKDIR/Percona-XtraDB-Cluster-5.7.33-rel36-49.1.Linux.x86_64.glibc2.12-minimal.tar.gz  https://www.percona.com/downloads/Percona-XtraDB-Cluster-57/Percona-XtraDB-Cluster-5.7.33-31.49/binary/tarball/Percona-XtraDB-Cluster-5.7.33-rel36-49.1.Linux.x86_64.glibc2.12-minimal.tar.gz
+
+# Installing PXC-5.7 client package
+    sudo percona-release enable pxc-57 release
+    sudo yum install -y Percona-XtraDB-Cluster-client-57
```

Comment:
we are running percona-release enable and yum/up install ny times. Why not create 2 functions and call them with params reponame and package name?

Reply:
Fixed